middle_end/compilenv.ml

```diff
@@ -104,8 +104,17 @@ let symbolname_for_pack pack name =
 
 let unit_id_from_name name = Ident.create_persistent name
 
+let no_leading_or_consecutive_underscores str =
+  String.mapi
+    (fun i c -> if c = '_' && (i = 0 || str.[i - 1] = '_') then '.' else c)
+    str
+
 let concat_symbol unitname id =
```

Comment:
I find that the name `concat_symbol` suggests that the operation can be repeated (`concat_symbol (concat_symbol foo bar) baz`), which makes the escaping scheme incorrect if I understand correctly. Could you maybe rename it into `absolute_symbol`, which I think clarifies that the operands should not themselves be the results of prefixing?